polar-core/src/bindings.rs

```diff
@@ -311,3 +318,19 @@ impl BindingManager {
     // variable_bindings
     // bindings
 }
+
+/// Get variable out of a term for ``add_constraint`` to determine where the
+/// constraint is stored.
+///
+/// If the term is a variable, the variable is returned.
+/// If the term is a dot expression, the VAR from VAR.field is returned.
+/// Otherwise, the term is returned.
+fn extract_variable(value: &Value) -> &Value {
```

Comment:
When does this happen? Are there cases when we call `add_constraint` on a term like:
`x.foo <= 1` ? Why wouldn't that be rewritten

Reply:
Removed in new code